### PR TITLE
feat: glob schema endpoint for generate

### PR DESCRIPTION
**Pull request by LiamMartens**
Using multiple gql files and combining them is a fairly common use case I believe.

### CODE REVIEW

File: packages/cli/src/inspectWriteGenerate.ts [hunk 1]
```diff
@@ -105,16 +107,24 @@ export async function inspectWriteGenerate({
...
+        )
+      );
 
       if (endpoint.endsWith('.json')) {
         const parsedFile:
           | IntrospectionQuery
           | { data?: IntrospectionQuery }
-          | undefined = JSON.parse(file);
+          | undefined = fileContents.reduce(
+          (acc, file) => merge(acc, JSON.parse(file)),
```
Comment (vicary) [inline packages/cli/src/inspectWriteGenerate.ts:125]:
```suggestion
          (acc, file) => ({
            ...acc,
            ...JSON.parse(file),
          }),
```

I see you are still working on it, but honestly lodash, or lodash.anything, does not worth the bundle bloat.

Do you think it works for the suggested change without lodash?

Comment (LiamMartens) [reply packages/cli/src/inspectWriteGenerate.ts]: Fine by me - made the change.
I don't personally know how this affects merging introspection JSONs but I also don't think anyone will use it with multiple JSONs - its more a use case of using multple .gql schema files anyways so I believe this should be fine

**Comment by vicary** [comment]
@LiamMartens Sorry I was not being clear enough. Some existing codebase, which I am planning to remove, is still using lodash. So further depending `lodash` itself, or adding `lodash.*`, is discouraged.

I think we should add `lodash` back to `package.json` to keep tests green before the above refactoring happens.

**Review by PabloSzx — CHANGES_REQUESTED** [review]
Accepting more than a single JSON as a single schema is an anti-pattern

Something like this has to be done:

- If every found file is a `.gql` or `.graphql`, merge them as it already is defined
- If a single file is found that is `.json`, use it at is already is
- If a combination of files extensions are found, (`.gql` | `.graphql`) + `.json`, throw an error
- If more than a single .json, throw an error

**Comment by LiamMartens** [comment]
@PabloSzx I updated it now to just allow gql or graphql, maybe it makes sense to just support it like that as there is really no proper use case for a JSON glob

**Review by PabloSzx — CHANGES_REQUESTED** [review]
thank you @LiamMartens, looks nice, the only thing missing seems to be new tests for the validations added, and you can also add the changeset, here is a quick description of how to do it https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md

**Review by PabloSzx — APPROVED** [review]
Thank you @LiamMartens :tada:

**Pull request merged**